Loop/View Models/BolusEntryViewModel.swift

```diff
@@ -193,6 +196,19 @@ final class BolusEntryViewModel: ObservableObject {
             return
         }
 
+        // Authenticate the bolus before saving anything
+        if enteredBolus.doubleValue(for: .internationalUnit()) > 0 {
+            let message = String(format: NSLocalizedString("Authenticate to Bolus %@ Units", comment: "The message displayed during a device authentication prompt for bolus specification"), enteredBolusAmountString)
+            authenticate(message) {
+                switch $0 {
+                case .success: self.continueSaving(onSuccess: completion)
+                case .failure: break
+                }
+            }
+        }
+    }
+    
+    private func continueSaving(onSuccess completion: @escaping () -> Void) {
```

Comment:
The check for the manual BG should definitely be moved, although the reason that it calls `continueSaving` after the authentication succeeds is so that we don't have to manage different threads. I could just have it `wait` until the authentication is done - thoughts?

Reply:
The order of `authenticate` and `continueSaving` is fine, but you are right, the manual glucose sample check needs to be moved somewhere before `authenticate`.